### PR TITLE
ARROW-6912: [Java] Extract a common base class for avro converter consumers

### PR DESCRIPTION
**Pull request by tianchen92**
Related to [ARROW-6912](https://issues.apache.org/jira/browse/ARROW-6912).

Currently Avro converter consumers have some common variables and methods which could be eliminated by extracting a common class.

### CODE REVIEW

**Comment by tianchen92** [comment]
@emkornfield Hi, Micah, please take a look when you have time, thanks!

File: java/adapter/avro/src/main/java/org/apache/arrow/consumers/BaseAvroConsumer.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.consumers;
+
+import org.apache.arrow.vector.FieldVector;
+
+/**
+ * Base class for all avro consumers.
```
Comment (emkornfield) [inline java/adapter/avro/src/main/java/org/apache/arrow/consumers/BaseAvroConsumer.java]: this doesn't include skipping consumers though?  Update the comment?

Comment (tianchen92) [reply java/adapter/avro/src/main/java/org/apache/arrow/consumers/BaseAvroConsumer.java]: yes, updated.

**Comment by tianchen92** [comment]
@emkornfield could we merge this? I have a follow-up PR for logical types.

**Comment by emkornfield** [comment]
+1, thanks.

**Pull request closed without merging**